tests/pass-dep/libc/libc-strerror_r.rs

```diff
@@ -0,0 +1,13 @@
+//@ignore-target: windows # Supported only on unixes
+
+fn main() {
+    unsafe {
+        let mut buf = vec![0u8; 32];
+        assert_eq!(libc::strerror_r(libc::EPERM, buf.as_mut_ptr().cast(), buf.len()), 0);
+        let mut buf2 = vec![0u8; 64];
+        assert_eq!(libc::strerror_r(-1i32, buf2.as_mut_ptr().cast(), buf2.len()), 0);
+        let mut buf3 = vec![0u8; 32];
+        // we purposely fill not enough length of the buffer to trigger ERANGE.
```

Comment:
```suggestion
        // This buffer is deliberately too small so this triggers ERANGE.
        let mut buf3 = vec![0u8; 32];
```

Reply:
You didn't apply my suggestion. The comment should be above the `let mut`, not below -- it's odd for a comment to refer to things above it.

Reply:
well because it refers to the fact I use.2 bytes off 32, the buffer itself is large enough it is just "misused".

Reply:
it was just an example but whatever..